base/io.jl

```diff
@@ -499,11 +499,18 @@ read!(filename::AbstractString, a) = open(io->read!(io, a), convert(String, file
     readuntil(stream::IO, delim; keep::Bool = false)
     readuntil(filename::AbstractString, delim; keep::Bool = false)
 
+    # Check if elements in the vector are of the right type
+    if !(all(isequal(eltype(delim), UInt8) || issubclass(eltype(delim), AbstractChar) for d in delim))
+        throw(ArgumentError("readuntil: elements in the vector must be of type UInt8 or AbstractChar"))
+    end
+
```

Comment:
This code is inside a docstring, and is therefore not executed.
Maybe instead, you could declare the type of the argument in the function signature to be `::Union{UInt8, AbstractChar, AbstractString, AbstractVector{UInt8}, AbstractVector{<:AbstractChar}}`?

You would then also need to check the function, and all the functions the function calls, to make sure it can handle each of these 5 argument types.
Perhaps also write tests for all five argument types.

Reply:
Something like this :   
# Arguments
- `stream::IO`: Input stream.
- `filename::AbstractString`: Name of the file.
- `delim::Union{UInt8, AbstractChar, AbstractString, AbstractVector{UInt8}, AbstractVector{<:AbstractChar}}`: Delimiter.
- `keep::Bool = false`: Controls whether the delimiter is included in the result.

Reply:
No, what I mean is: Instead of merely _writing_ what type the `delim` keyword should be in the docstring, enforce it, by changing the type in the function signature.

Reply:
Should all changes be in docstring?